src/python/WMCore/MicroService/MSPileup/MSPileupTasks.py

```diff
@@ -100,17 +105,170 @@ def pileupSizeTask(self):
+            # - when container fraction is increasing: new container is a superset of customName + subset of pileupName
+
+            # usage of block names defined in this logic:
+            # https://github.com/dmwm/WMCore/pull/11807#pullrequestreview-1786778783
+            customBlocks = self.getPileupBlocks(doc, previousFraction, fraction)
+
+            # create new container DID in Rucio for our custom name
+            self.rucioClient.createContainer(doc['customName'], scope=self.customRucioScope)
+            self.logger.info("Create container %s with scope %s", doc['customName'], self.customRucioScope)
+
+            # call rucio APIs to attach custom blocks to our custom container (DID)
+            newRules = []
+            self.rucioClient.attachDIDs(None, doc['customName'], customBlocks, scope=self.customRucioScope)
+            for rse in doc['expectedRSEs']:
+                # create new rule for custom DID using pileup document rse
+                newRules += self.rucioClient.createReplicationRule(doc['customName'], rse)
+
+            self.logger.info("Custom pileup: %s has the following new rules created: %s for RSEs: %s",
+                             doc['customName'], newRules, doc['expectedRSEs'])
+            # set expiration date (to be 24h) for already existing ruleIds from pileup document
+            for rid in doc['ruleIds']:
+                # set expiration date to be 24h ahead of right now
+                opts = {'lifetime': 24 * 60 * 60}
+                self.rucioClient.updateRule(rid, opts)
+                self.logger.info("Custom pileup %s has new lifetime %s", doc['customName'], opts['lifetime'])
+
+            # update pileup document
+            if newRules:
+                # remove duplicate rules
+                newRules = set(newRules)
+                try:
+                    # update rules
+                    doc['ruleIds'] += newRules
+
+                    # update transition record of the pileup document, see logic:
+                    # https://gist.github.com/amaltaro/b4f9bafc0b58c10092a0735c635538b5
+
+                    # get previous transition record
+                    prevTranRecord = doc['transition'][-1]
+
+                    # keep custom name value from this record as we'll update it later
+                    cname = prevTranRecord['customDID']
+
+                    # update previous transition record in place, i.e. it will be updated
+                    # within doc['transition'] directly after these assingments
+                    prevTranRecord['containerFraction'] = fraction
+                    prevTranRecord['customDID'] = cname
+                    prevTranRecord['updateTime'] = gmtimeSeconds()
+
+                    # update custom name from value of previous transition record
+                    doc['customName'] = cname
+                    self.logger.info("update pileup document %s", doc)
```

Comment:
Given that this log record will print the full document, similar to 3 lines below, shouldn't we instead log something like "Finally, updating the pileup document for pileup name: %s"?

Reply:
ok